quickwit/quickwit-metastore/src/metastore/file_backed/file_backed_index/mod.rs

```diff
@@ -213,6 +213,11 @@ impl FileBackedIndex {
         &self.metadata
     }
 
+    /// Mutable ref to index config.
+    pub fn index_config_mut(&mut self) -> &mut IndexConfig {
+        &mut self.metadata.index_config
+    }
```

Comment:
if I understand correctly, it is okay to mutate any part of the `IndexMetadata`. But I wouldn't mind a second opinion from the reviewer.